src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let title = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0) {
+        title += separator;
+      };
+
+      if (this[i] !== null
+      && this[i] !== undefined) {
+        title += this[i];
+      };
```

Comment:
Could be simplified using [Nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing).